contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -129,34 +129,63 @@ IECoreScene::PointsPrimitivePtr convertPrimitive( pxr::UsdGeomPoints points, pxr
 	IECore::V3fVectorDataPtr positionData = DataAlgo::fromUSD( pointsData );
 	positionData->setInterpretation( GeometricData::Point );
-
 	IECoreScene::PointsPrimitivePtr newPoints = new IECoreScene::PointsPrimitive( positionData );
 
-	IECore::DataPtr orientationData = DataAlgo::fromUSD( pointInstancer.GetOrientationsAttr(), time );
-	newPoints->variables["orient"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, orientationData );
+	// Per point attributes
 
-	IECore::DataPtr instanceIndicesData = DataAlgo::fromUSD( pointInstancer.GetProtoIndicesAttr(), time );
-	newPoints->variables["instanceIndex"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, instanceIndicesData );
+	if( auto protoIndicesData = DataAlgo::fromUSD( pointInstancer.GetProtoIndicesAttr(), time ) )
+	{
+		newPoints->variables["prototypeIndex"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, protoIndicesData );
+	}
 
-	IECore::DataPtr scalesData = DataAlgo::fromUSD( pointInstancer.GetScalesAttr(), time );
-	newPoints->variables["scale"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, scalesData );
+	if( auto idsData = DataAlgo::fromUSD( pointInstancer.GetIdsAttr(), time ) )
+	{
+		newPoints->variables["instanceId"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, idsData );
+	}
+
+	if( auto orientationData = DataAlgo::fromUSD( pointInstancer.GetOrientationsAttr(), time ) )
+	{
+		newPoints->variables["orientation"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, orientationData );
+	}
+
+	if( auto scaleData = DataAlgo::fromUSD( pointInstancer.GetScalesAttr(), time ) )
+	{
+		newPoints->variables["scale"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, scaleData );
+	}
+
+	if( auto velocityData = DataAlgo::fromUSD( pointInstancer.GetVelocitiesAttr(), time ) )
+	{
+		newPoints->variables["velocity"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, velocityData );
+	}
+
+	if( auto accelerationData = DataAlgo::fromUSD( pointInstancer.GetAccelerationsAttr(), time ) )
+	{
+		newPoints->variables["acceleration"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, accelerationData );
+	}
+
+	if( auto angularVelocityData = DataAlgo::fromUSD( pointInstancer.GetAngularVelocitiesAttr(), time ) )
+	{
+		newPoints->variables["angularVelocity"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Vertex, angularVelocityData );
+	}
+
+	// Prototype paths
+
+	pxr::SdfPathVector targets;
+	pointInstancer.GetPrototypesRel().GetTargets( &targets );
+
+	IECore::StringVectorDataPtr prototypeRootsData = new IECore::StringVectorData();
+	auto &prototypeRoots = prototypeRootsData->writable();
+	prototypeRoots.reserve( targets.size() );
+	for( const auto &t : targets )
+	{
+		prototypeRoots.push_back( t.GetString() );
+	}
```

Comment:
It could, but as far as I know this is the only place we need this very specific conversion from `SdfPathVector`. Until something can is useful publicly, it's simpler to keep it private, and gives us more options for changing things. Note that even DataAlgo is private in that we're not publishing the header, but I intend to change that in a future PR.

Reply:
Fair enoguh, USD relationships get used a fair bit IIRC, but we can sort that when we encounter them again...